src/NzbDrone.Core/Datastore/Migration/204_add_release_hash.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using Dapper;
+using FluentMigrator;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Datastore.Migration.Framework;
+using NzbDrone.Core.Parser.Model;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(204)]
+    public class add_add_release_hash : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Alter.Table("EpisodeFiles").AddColumn("ReleaseHash").AsString().Nullable();
+
+            Execute.WithConnection(UpdateEpisodeFiles);
+        }
+
+        private void UpdateEpisodeFiles(IDbConnection conn, IDbTransaction tran)
+        {
+            var updates = new List<object>();
+
+            using (var cmd = conn.CreateCommand())
+            {
+                cmd.Transaction = tran;
+                cmd.CommandText = "SELECT \"Id\", \"SceneName\", \"RelativePath\", \"OriginalFilePath\" FROM \"EpisodeFiles\"";
+
+                using var reader = cmd.ExecuteReader();
+                while (reader.Read())
+                {
+                    var id = reader.GetInt32(0);
+                    var sceneName = reader[1] as string;
+                    var relativePath = reader[2] as string;
+                    var originalFilePath = reader[3] as string;
+
+                    ParsedEpisodeInfo parsedEpisodeInfo = null;
+
+                    var originalTitle = sceneName.IsNotNullOrWhiteSpace() ? sceneName : (originalFilePath.IsNotNullOrWhiteSpace() ? Path.GetFileNameWithoutExtension(originalFilePath) : null);
```

Comment:
Since `sceneName` is null when it's not set this can be shortened to:

```suggestion
                    var originalTitle = sceneName ?? originalFilePath.IsNotNullOrWhiteSpace() ? Path.GetFileNameWithoutExtension(originalFilePath) : null;
```

Reply:
Can SceneName ever be the empty string in the database?
I went the IsNullOrWhiteSpace route since that's what FileNameBuilder did.

Reply:
Should only be only be null or a valid value unless someone manually altered their DB to set it to empty string.

Yeah, we use `Is(Not)NullOrWhiteSpace` across the app to avoid having to check bot, but in the migration just the null check is fine.

Reply:
Thinking about it a bit more, not much harm in the empty string check, so this would be what I'd do to avoid the double ternary.

```suggestion
                    var originalTitle = sceneName;
                    
                     if (originalTitle.IsNullOrWhiteSpace() && originalFilePath.IsNotNullOrWhiteSpace())
                     {
                         originalTitle = Path.GetFileNameWithoutExtension(originalFilePath);
                     }
```